src/main/java/com/openshift/internal/restclient/model/ReplicationController.java

```diff
@@ -99,6 +99,44 @@ public void setEnvironmentVariable(String containerName, String name, String val
 	}
 
 	@Override
+	public void removeEnvironmentVariable(String name) {
+		removeEnvironmentVariable(null, name);
+	}
+
+	@Override
+	public void removeEnvironmentVariable(String containerName, String name) {
+		String defaultedContainerName = StringUtils.defaultIfBlank(containerName, "");
+		ModelNode specContainers = get(SPEC_TEMPLATE_CONTAINERS);
+		if(specContainers.isDefined()) { //should ALWAYS exist
+			List<ModelNode> containers = specContainers.asList();
+			if(!containers.isEmpty()) {
+
+				Optional<ModelNode> opt = containers.stream().filter(n->defaultedContainerName.equals(asString(n, NAME))).findFirst();
+				ModelNode node =  opt.isPresent() ? opt.get() : containers.get(0);
+				ModelNode envNode = get(node, ENV);
+
+				List<ModelNode> varList = new ArrayList<>();
+				if (ModelType.LIST.equals(envNode.getType())){
+					varList.addAll(envNode.asList());
+				}
+
+				//Check if variable exists
+				Optional<ModelNode> targetVar = varList.stream().filter(n->name.equals(asString(n, NAME))).findFirst();
```

Comment:
Since null name will not do any good, can it be checked in the first line of the method? What is better, return silently or throw IllegalArgumentException?